### PR TITLE
JE- Adds Checkbox filter to AdminUserPage

### PR DESCRIPTION
**Pull request by JamesEllenstein**
Adds filter to allow Admin to filter table using checkboxes  by All, Riders, or drivers. Also updated test files. Change made to useBackendMutation to allow for multiple query keys to be invalidated.
Storybook: [Before](https://ucsb-cs156-s23.github.io/proj-gauchoride-s23-5pm-1/storybook/?path=/story/pages-adminuserspage-adminuserspage--default) [After](https://ucsb-cs156-s23.github.io/proj-gauchoride-s23-5pm-1/prs/48/storybook/?path=/story/pages-adminuserspage-adminuserspage--default)

### CODE REVIEW

**Review by zpr-ucsb — APPROVED** [review]
Looks good, seems like everything was added correctly.

File: frontend/src/main/utils/useBackend.js [hunk 1]
```diff
@@ -68,9 +68,12 @@ export function useBackendMutation(objectToAxiosParams, useMutationParams, query
         // Stryker disable all: Not sure how to set up the complex behavior needed to test this
         onSettled: () => {
             if (queryKey !== null)
-                queryClient.invalidateQueries(queryKey);
+                //if (Array.isArray(queryKey)) {
```
Comment (pconrad) [inline frontend/src/main/utils/useBackend.js:71]: Ooops... please remove commented out code

**Review by pconrad — DISMISSED** [review]
One line of commented out code.  Otherwise, LGTM

File: frontend/src/main/utils/useBackend.js (hunk 1 above)
Comment (JamesEllenstein) [reply frontend/src/main/utils/useBackend.js]: Done in most recent commit

**Review by vikram-elango — APPROVED** [review]
LGTM!

**Review by pconrad — APPROVED** [review]
LGTM!  WIll test on staff-qa

**Pull request merged**